zerver/migrations/0524_userpresence_last_update_id.py

```diff
@@ -0,0 +1,37 @@
+# Generated by Django 5.0.5 on 2024-05-02 02:17
+
+from django.contrib.postgres.operations import AddIndexConcurrently
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    atomic = False
+
+    dependencies = [
+        ("zerver", "0523_alter_multiuseinvite_subscribe_to_default_streams_and_more"),
+    ]
+
+    operations = [
+        # Create a new column, making it NULLable to avoid locking the table
+        # rewriting rows with a non-NULL default value.
+        migrations.AddField(
+            model_name="userpresence",
+            name="last_update_id",
+            field=models.PositiveBigIntegerField(db_index=True, null=True),
+        ),
+        # This is an SQL noop, since Django doesn't add defaults at database level.
+        # The default guarantees new rows will have a value. Old rows can get backfilled
+        # in the next migration.
+        migrations.AlterField(
+            model_name="userpresence",
+            name="last_update_id",
+            field=models.PositiveBigIntegerField(db_index=True, default=0, null=True),
+        ),
+        AddIndexConcurrently(
```

Comment:
A small optimization is that it should be more efficient to do the backfill first, and this AddIndexConcurrently after that, rather than vice versa.

That way the backfill doesn't have to churn the index while it's updating all the rows; effectively it means we write the index just once, instead of writing it and then writing over the whole thing.

Reply:
Oh yeah that's a nice idea, I moved this operation to the end of the `user_presence_backfill_last_update_id_to_0.py` migration